whylog/config/rule.py

```diff
@@ -35,6 +44,23 @@ def get_causes_parsers(self):
     def get_effect_name(self):
         return self._effect.name
 
+    def constraints_check(self, clues, effect_clues_dict):
+        """
+        check if given clues satisfy rule
+        basing on its causes, effect and constraints.
+        returns list of InvestigationResult objects
+        """
+        clues_lists = [
+            (clues[parser_name], occurrences)
+            for parser_name, occurrences in self._frequency_information if parser_name in clues
```

Comment:
lookup is done twice here. Use `clues.get(parser_name)` and check with `is None`